### PR TITLE
Modified restrictions on `margins` method

### PR DESCRIPTION
**Pull request by AtharvaKhare**
## PR Summary

Modified and added example for `matplotlib.axes.Axes.margins()`.

Fixes https://github.com/matplotlib/matplotlib/issues/9334.

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/tests/test_axes.py [hunk 1]
```diff
@@ -4593,13 +4593,13 @@ def test_margins():
...
-    assert ax2.margins() == (1, 0.5)
+    ax2.margins(0.5, 2)
+    assert ax2.margins() == (0.5, 2)
 
     fig3, ax3 = plt.subplots(1, 1)
     ax3.plot(data)
-    ax3.margins(x=1, y=0.5)
-    assert ax3.margins() == (1, 0.5)
+    ax3.margins(x=-0.2, y=0.5)
+    assert ax3.margins() == (-0.2, 0.5)
```
Comment (jklymak) [inline lib/matplotlib/tests/test_axes.py]: I think these tests would be better if they also asserted the x-lim and y-lim you expect.  i.e. `assert ax3.get_xlim() == (left, right)`.  That way if anyone monkeys with the internal code this test will catch the inconsistency.

Comment (AtharvaKhare) [reply lib/matplotlib/tests/test_axes.py]: Pushed 76a2721.

**Comment by jklymak** [comment]
#9463 should fix your docs-python errors. When that gets accepted, you can merge w/ master and this should be good to go if the tests pass (#9463 might take a couple of days).  If you need help merging with master, folks here can help.

**Comment by AtharvaKhare** [comment]
Can someone trigger rebuild for circleci?
Is there anything additional to be done by me?

**Comment by jklymak** [comment]
You’ll need to merge the upstream master with your branch.  Basically follow the instructions at https://help.github.com/articles/syncing-a-fork/ but where they tell you to checkout master you will checkout flexible-axis-margins.  Then push back to github.

**Comment by tacaswell** [comment]
Who ever merges this please squash-merge to get rid of the merge from master (or @AtharvaKhare can you rebase this on to current master?).

**Comment by jklymak** [comment]
@AtharvaKhare  we can squash-merge if you like.  This makes your 6 commits look like one commit and makes the matplotlib commit history much cleaner.

If you'd like yet another bit of git-fu, you can `git rebase` the two commits into one.  I usually do a `git log` and count how many commits I want to merge (say 6 in your case) and do `git rebase -i HEAD~6` (though I always forget if it should be 6 or 7).  Then in the editor that should pop up I "squash" the newest entries and pick the oldest entry.   You then edit to edit the log entry. 

Then you need to `push origin flexible-axis-margins --force` to put the new single commit onto github.  You need to `--force` because you are changing history.  

There are many guides to this online.  http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html.

I sometimes screw it up.  A good idea is to make a backup of your repository, then you can start again if you need to.  `push --force` will always overwrite any old mistakes you made.

If you have trouble just ask.  This is kind of an optipnal step on your part, but I think "Squash and merge" from our end is risky in case of conflicts...


[as an aside, none of this is actually in our development docs.  It might be reasonable to add a short guide outlining the basic git workflows]

**Comment by tacaswell** [comment]
It was in our development docs for a while and then was removed by updating gitwash, see https://github.com/matthew-brett/gitwash/pull/13

File: examples/axes_grid1/axes_margins.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+"""
+============
+Margins demo
+============
+
```
Comment (NelleV) [inline examples/axes_grid1/axes_margins.py]: Can you please add a small paragraph explaining what the example does?
The title is also not very informative.

Comment (AtharvaKhare) [reply examples/axes_grid1/axes_margins.py]: Pushed https://github.com/matplotlib/matplotlib/pull/9459/commits/e3280c67845040bc1926d13d1160783a3d6ab74c. Is it enough or should I elaborate?

**Comment by jklymak** [comment]
@tacaswell  Ah OK, it is indeed at http://matplotlib.org/devel/gitwash/development_workflow.html

I admit that I find the heading "Working with Matplotlib source code" a bit obscure and didn't think to check there on a quick skim.  I'd have said "Using git and github with Matplotlib source code".

File: examples/axes_grid1/axes_margins.py (hunk 2 above)
Comment (NelleV) [reply examples/axes_grid1/axes_margins.py]: Do you mind if I push some changes directly to the branch?

Comment (AtharvaKhare) [reply examples/axes_grid1/axes_margins.py]: Go ahead.

**Comment by AtharvaKhare** [comment]
@NelleV Thank you!
@tacaswell Do the commits need to be squashed again?

**Comment by choldgraf** [comment]
does it still need a squash and merge?

**Pull request merged**